core/controllers/feedback.py

```diff
@@ -108,3 +114,66 @@ def get(self):
             'cursor': new_urlsafe_start_cursor,
             'more': more,
         })
+
+
+class SuggestionHandler(base.BaseHandler):
+    """"Handles operations relating to learner suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+
+    @base.require_user
+    def post(self, exploration_id):
+        feedback_services.create_suggestion(
+            exploration_id,
+            self.user_id,
+            self.payload.get('exploration_version'),
+            self.payload.get('state_name'),
+            self.payload.get('suggestion_content'))
+        self.render_json(self.values)
+
+
+class SuggestionActionHandler(base.BaseHandler):
+    """"Handles actions performed on threads with suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+    ACCEPT_ACTION = 'accept'
+    REJECT_ACTION = 'reject'    
+
+    @base.require_user
+    def put(self, exploration_id, thread_id):
+        action = self.payload.get('action')
+        if action.lower() == self.ACCEPT_ACTION.lower():
```

Comment:
No -- we should still check that the action is valid. If the action is neither 'accept' or 'reject' then this is grounds for raising an exception, even if the word is the same but with different casing.